### ISSUE TITLE
Add Python 3.11 final to CI

### ISSUE THREAD

**Issue by parafoxia** [parafoxia/analytix#42]
### Summary

`3.11-dev` should be changed to `3.11` in the CI workflow.

~~This is currently blocked by https://github.com/actions/runner-images/issues/6459.~~ The Darwin build artefact is now available.

### Why?

Python 3.11 final has been released.

### Contribution

- [X] I wish to contribute this feature.

### Checklist

- [X] I have searched the issue tracker for duplicate issues and couldn't find any. If this is a follow-up issue, I have specified as such.
